projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -397,6 +397,13 @@ export class GridBaseAPIService<T extends GridType> implements GridServiceType {
         if (args.cancel) {
             return;
         }
+
+        const isHierarchicalGrid = grid.nativeElement.tagName.toLowerCase() === 'igx-hierarchical-grid';
```

Comment:
If we ever turn this into a web component and use a different tag name, this won't function. I suppose a different way should be figured to recognize a type of grid inside the api service from a tag name check.

Reply:
@skrustev indeed. If the service needs to know what type of grid it is instantiated for, this can be injected as a token or input/property.

Reply:
For now one way this could be resolved even without properties is to introduce `endEdit` method call in `set_row_expansion_state` in the `gridAPI` and override it in the `hgridAPI to call `endEditAll` in it as well. 
For the other similar cases of `'igx-hierarchical-grid' can be logged separately.

Reply:
@skrustev That's what I was also thinking. It should have an override. I was looking at the code today to see the more general picture and there's a bunch of things that need to be done - `hgridAPI` and `gridAPI` being two references to one and the same thing being one of them...